offlinevalidator/offlinevalidator.go

```diff
@@ -87,11 +85,51 @@ func Validate(context *context.Context) bool {
 	resources := obtainResources(deNilizedTemplate, perunTemplate, context.Logger)
 	deadResources := getNilResources(resources)
 	deadProperties := getNilProperties(resources)
+	hasAllowedValues := findingAllowedValues(goFormationTemplate, context.Logger)
+	if hasAllowedValues == true {
+		return false
+	}
 
 	valid = validateResources(resources, &specification, context.Logger, deadProperties, deadResources)
 	return valid
 }
 
+func findingAllowedValues(goFormationTemplate cloudformation.Template, logger *logger.Logger) bool {
+	parameters := goFormationTemplate.Parameters
+	isType := false
+	isAllovedValues := false
+
+	for _, value := range parameters {
+		valueof := reflect.ValueOf(value)
+		if valueof.Kind() == reflect.Map {
```

Comment:
1. Can this value be any other kind? From documentation I think it can not. If we want to validate the structure, should this function be responsible for that? I believe that this function purpose was only to check if for allowedValues there is no other type than String.
2. If we made sure it is a map as value, can't we access it directly, making it a lot simpler than through reflection?
something like 
if value["AllowedValues"] != nil && value["Type"] != "String" { logger.Error.. } 
wouldn't it be sufficient?
3. Function name should indicate its purpose, here I believe it should be something like hasAllowedValuesParametersValid and the return would be opposite.

Reply:
1. and 3. I will improve it. 
2. To use map I have to decode `parameters` two times. Then I have problem with validation - sometimes it returns true or false. So I decide to use `reflect` and type `reflect.Value`.